testutil/daemon/daemon.go

```diff
@@ -140,13 +140,51 @@ func New(t testingT, ops ...func(*Daemon)) *Daemon {
 		dockerdBinary:   defaultDockerdBinary,
 		swarmListenAddr: defaultSwarmListenAddr,
 		SwarmPort:       DefaultSwarmPort,
-		log:             t,
+		log:             nopLog{},
 	}
 
 	for _, op := range ops {
 		op(d)
 	}
 
+	return d, nil
+}
+
+// NewDaemon returns a Daemon instance to be used for testing.
+// This will create a directory such as d123456789 in the folder specified by
+// $DOCKER_INTEGRATION_DAEMON_DEST or $DEST.
+// The daemon will not automatically start.
+func NewDaemon(ops ...func(*Daemon)) (*Daemon, error) {
```

Comment:
Leave it so that the `NewDaemon` takes in the dest ("root" is probably better name) as param. Using env to pass values to public methods doesn't make much sense. And you can pass the logger as well or make it settable through ops. Default or nil can be the `nopLog`.

Reply:
> Leave it so that the NewDaemon takes in the dest ("root" is probably better name) as param.

I was trying to be consistent, but thinking about it more I completely agree. If this isn't for tests it shouldn't use the same environment variables. I can remove where this env variable is duplicated and used in buildkit too, which should remove a potential point where tests could easily be broken.

I also added a setting that lets you pass in a test to log to. This could be more generic, but that would require exporting the `logT` interface. I could go either way if you think other types of loggers will be required in the future, but I think it makes sense to only make the minimum required change now and add another option if needed in the future.